BHoMAnalytics_UI/CaptureProjectData.xaml.cs

```diff
@@ -43,57 +43,132 @@ namespace BH.UI.Analytics
+            {
+                GrassHopperUIText.Visibility = Visibility.Visible;
+            }
+            if (_uiName == "Revit")
+            {
+                BHoMUseTextBlock.Text = "Please indicate purpose of this use of BHoM Tools for Revit";
+            }
+
             ProjectBtn.Focus();
+            this.ShowDialog();
         }
+        #endregion
+
+        #region Properties
+        public string Version;
+        public string _uiName;
+
+        public int _initialHeight = 240;
+        #endregion
 
+        #region EventActions
+        //Project
         private void Click_ProjectBtn(object sender, EventArgs e)
         {
+            ResetForms();
             ProjectInputPanel.Visibility = Visibility.Visible;
-            this.Height = 250;
+            this.Height = 380;
             ProjectIDInput.Focus();
         }
+        private void Click_ConfirmProjectBtn(object sender, EventArgs e)
+        {
+            ConfirmProject();
+        }
+        private void KeyDown_ProjectIDEntry(object sender, KeyEventArgs e)
+        {
+            if (e.Key == Key.Enter || e.Key == Key.Return)
+                ConfirmProject();
+        }
 
+        //Non Project
         private void Click_NonProjectBtn(object sender, EventArgs e)
         {
-            string projectID = "Non-Project work";
-
-            BH.Engine.Base.Compute.RecordEvent(new ProjectIDEvent
-            {
-                Message = "The project ID for this file is now set to " + projectID,
-                ProjectID = projectID
-            });
-
-            this.Close();
+            ResetForms();
+            NonProjectBtn.Focus();
+            NonProjectSelectionPanel.Visibility = Visibility.Visible;
+            this.Height = 380;
+            NonProjectListBox.SelectedIndex = 0;
+            NonProjectListBox.Focus();
         }
 
-        private void Click_ConfirmProjectBtn(object sender, EventArgs e)
+        private void Click_ConfirmNonProjectBtn(object sender, RoutedEventArgs e)
         {
             ConfirmProject();
         }
 
-        private void KeyDown_ProjectIDEntry(object sender, KeyEventArgs e)
+        private void KeyDown_NonProjectListBox(object sender, KeyEventArgs e)
         {
             if (e.Key == Key.Enter || e.Key == Key.Return)
                 ConfirmProject();
         }
 
+        private void NonProjectListBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (NonProjectListBox.SelectedValue.Equals(NonProjectOption.Other))
+            {
+                this.Height = 445;
+                OtherSpecifyInput.Visibility = Visibility.Visible;
+                OtherSpecifyText.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                this.Height = 380;
+                OtherSpecifyInput.Visibility = Visibility.Collapsed;
+                OtherSpecifyText.Visibility = Visibility.Collapsed;
+            }
+        }
+        //Window
```

Comment:
```suggestion
        }
        
        //Window
```

Same comment regarding the code comment